app/services/analytics_events.rb

```diff
@@ -248,4 +248,65 @@ def proofing_address_result_missing
   def proofing_document_result_missing
     track_event('Proofing Document Result Missing')
   end
+
+  # @identity.idp.event_name Return to SP: Failed to proof
+  # Tracks when a service provide fails to proof.
+  # @param [String] redirect_url the url of the service provider
+  def return_to_sp_failure_to_proof(url, **location_params)
```

Comment:
1. can we make `url` keyword argument to match?
2. looks like we can rely on `location_params` to have fairly consistent keys, let's use those? https://github.com/18F/identity-idp/blob/bd1e520eead1c854cd391a6a0448da73f037b00c/app/controllers/redirect/redirect_controller.rb#L3

```suggestion
  def return_to_sp_failure_to_proof(url:, flow: nil, step: nil, location: nil, **extra)
```